### PR TITLE
planner: refactor the cost implementation for PointGet/BatchGet

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #33945

Problem Summary: planner: refactor the cost implementation for PointGet/BatchGet

### What is changed and how it works?

planner: refactor the cost implementation for PointGet/BatchGet

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/find_best_task.go [hunk 1]
```diff
@@ -1982,9 +1983,11 @@ func (ds *DataSource) convertToPointGet(prop *property.PhysicalProperty, candida
 		pointGetPlan.IndexValues = candidate.path.Ranges[0].LowVal
 		pointGetPlan.PartitionInfo = partitionInfo
 		if candidate.path.IsSingleScan {
-			cost = pointGetPlan.GetCost(candidate.path.IdxCols)
+			pointGetPlan.accessCols = candidate.path.IdxCols
+			cost = pointGetPlan.GetCost()
```
Comment (ngaut) [inline planner/core/find_best_task.go:1987]: The same logic in both if/else branch, please move it out to make it clear.

File: planner/core/find_best_task.go [hunk 2]
```diff
@@ -2057,9 +2061,11 @@ func (ds *DataSource) convertToBatchPointGet(prop *property.PhysicalProperty,
 			batchPointGetPlan.Desc = prop.SortItems[0].Desc
 		}
 		if candidate.path.IsSingleScan {
-			cost = batchPointGetPlan.GetCost(candidate.path.IdxCols)
+			batchPointGetPlan.accessCols = candidate.path.IdxCols
+			cost = batchPointGetPlan.GetCost()
```
Comment (ngaut) [inline planner/core/find_best_task.go:2065]: Ditto

File: planner/core/find_best_task.go (hunk 1 above)
Comment (qw4990) [reply planner/core/find_best_task.go]: Fixed, PTAL

**Review by Reminiscent — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**